File: sdk/python/rustchain_sdk/bottube/exceptions.py
"""
BoTTube SDK Exceptions
"""

from typing import Optional


class BoTTubeError(Exception):
    """Base exception for BoTTube SDK"""
    pass


class AuthenticationError(BoTTubeError):
    """Authentication related errors"""
    pass


class APIError(BoTTubeError):
    """API request errors"""
    def __init__(self, message: str, status_code: Optional[int] = None, endpoint: Optional[str] = None):
        super().__init__(message)
        self.status_code = status_code
        self.endpoint = endpoint


class UploadError(BoTTubeError):
    """Video upload related errors"""
    def __init__(self, message: str, validation_errors: Optional[list] = None):
        super().__init__(message)
        self.validation_errors = validation_errors or []

File: sdk/javascript/bottube-sdk/src/exceptions.ts
/**
 * BoTTube SDK Exceptions
 */

export class BoTTubeError extends Error {
  constructor(message: string) {
    super(message);
    this.name = "BoTTubeError";
  }
}

export class AuthenticationError extends BoTTubeError {
  constructor(message: string) {
    super(message);
    this.name = "AuthenticationError";
  }
}

export class APIError extends BoTTubeError {
  statusCode?: number;
  endpoint?: string;

  constructor(message: string, statusCode?: number, endpoint?: string) {
    super(message);
    this.name = "APIError";
    this.statusCode = statusCode;
    this.endpoint = endpoint;
  }
}

export class UploadError extends BoTTubeError {
  validationErrors?: string[];

  constructor(message: string, validationErrors?: string[]) {
    super(message);
    this.name = "UploadError";
    this.validationErrors = validationErrors || [];
  }
}
